### DISCUSSION TITLE
RichLog doesn't want to render DataTable

### DISCUSSION THREAD

**Discussion by unwize** [Textualize/textual#3947] [Q&A] [answered]
When attempting to print a `DataTable `to `RichLog`, `RichLog `litterally prints `"DataTable()"` instead of rendering the widget. For reference, I followed [this tutorial](https://textual.textualize.io/widgets/rich_log/#__tabbed_1_2)

![image](https://github.com/Textualize/textual/assets/83423830/5673f05e-c2d0-4b86-b0a8-59aad5c72e15)

The code in question:

```

 @property
  def game_screen(self) -> RichLog:
      return self.query_one("#game_content_view", RichLog)
```

```
 self.game_screen.clear()
  self.game_screen.write(text)

  if print_table:
      table = DataTable()
      table.add_columns(*cols)
      table.add_rows(enumerate((rows))
      self.game_screen.write(table)
```

To me, this appears to be very similar to the example code linked above. The `cols`, `rows`, and `print_table` values are valid.

**Comment by davep** [comment, marked as answer]
When you say `DataTable`, do you mean the [Textual `DataTable` widget](https://textual.textualize.io/widgets/data_table/)? If so that isn't going to work. As the tutorial you point to says:

> Call [RichLog.write](https://textual.textualize.io/widgets/rich_log/#textual.widgets._rich_log.RichLog.write) with a string or [Rich Renderable](https://rich.readthedocs.io/en/latest/protocol.html) to write content to the end of the RichLog.

A `DataTable` is neither a string nor a Rich renderable; it's a Textual widget.

**Reply by unwize** [reply]
Gotcha, thanks.
